### PR TITLE
Added Tenant API

### PR DESCRIPTION
**Pull request by aabeshov**
### Description
Added Tenant API

### Issues Resolved
[#106 ]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: model/common_structures.smithy [hunk 1]
```diff
@@ -59,3 +59,29 @@ structure Hits{
...
+
+list PatchOperationList{
+    member: PatchOperation
+}
+
+structure PatchOperation {
+    op: String,
+    path: String,
+    value: AttributeMap
+}
```
Comment (nhtruong) [inline model/common_structures.smithy:87]: This file will grown extremely large if we put every structure here. It's only meant for structure to be reused like the results of different types of search. Can we move these into the structures.smithy file of the respective operation?

File: model/plugins/create_tenant/operations.smithy [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+$version: "2"
+namespace OpenSearch
+use opensearch.openapi#vendorExtensions
+
+@externalDocumentation(
+    "API Reference": "https://opensearch.org/docs/2.7/security/access-control/api/#create-tenant"
+)
+
+@vendorExtensions(
+    "x-operation-group": "create_tenant",
```
Comment (nhtruong) [inline model/plugins/create_tenant/operations.smithy:16]: remember to put this in `security` namespace. Same with other operations

Comment (nhtruong) [reply model/plugins/create_tenant/operations.smithy]: just operations of the security plugin

**Review by nhtruong — APPROVED** [review]

File: model/security/get_tenant/operations.smithy [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+@suppress(["HttpUriConflict"])
+@http(method: "GET", uri: "/_plugins/_security/api/tenants/")
+@documentation("List of all tenants.")
+operation GetTenants {
+    input: GetTenants_Input,
+    output: GetTenants_Output
+}
+
+@vendorExtensions(
+    "x-operation-group": "security.get_tenant",
```
Comment (nhtruong) [inline model/security/get_tenant/operations.smithy:29]: this [model/security/get_tenant/](https://github.com/opensearch-project/opensearch-api-specification/pull/113/files#diff-774b3f091e8939f7ab874e3c81bf18409f1d6c13bb002fc529937a68f820e810) folder actually hosts 2 different operation groups: `security.get_tenants` and `security.get_tenant`. There should only be one group per folder. And don't forget to add  the correct "API Reference" for each.

Comment (nhtruong) [reply model/security/get_tenant/operations.smithy:29]: Note that these 2 groups might share some common structures in the bodies. You only need to define it in one. Say define `Tenant` in `security/get_tenant` and you can reuse it in `security/get_tenants`

File: model/security/get_tenant/operations.smithy [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "x-version-added": "1.0",
+)
+@readonly
+@suppress(["HttpUriConflict", "HttpMethodSemantics.UnexpectedPayload"])
+@http(method: "GET", uri: "/_plugins/_security/api/tenants/{tenant}")
+@documentation("Returns information about given tenant.")
```
Comment (nhtruong) [inline model/security/get_tenant/operations.smithy:35]: Let's stick the language used in the api reference: `Retrieves one tenant.`

File: model/security/patch_tenant/operations.smithy [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "x-version-added": "1.0",
+)
+@idempotent
+@suppress(["HttpUriConflict"])
+@http(method: "PATCH", uri: "/_plugins/_security/api/tenants/")
+@documentation("Add, delete, or modify multiple tenants in a single call.")
+operation PatchTenants {
+    input: PatchTenants_Input,
+    output: PatchTenants_Output
+}
```
Comment (nhtruong) [inline model/security/patch_tenant/operations.smithy:39]: Similar issues found in [model/security/get_tenant/operations.smithy](https://github.com/opensearch-project/opensearch-api-specification/pull/113/files#diff-774b3f091e8939f7ab874e3c81bf18409f1d6c13bb002fc529937a68f820e810)

File: model/security/security.smithy [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+//
+//  The OpenSearch Contributors require contributions made to
+//  this file be licensed under the Apache-2.0 license or a
+//  compatible open source license.
+
+$version: "2"
+namespace OpenSearch
+
+@documentation("The message indicating the information of action of the model.")
+string Message
```
Comment (aabeshov) [inline model/security/security.smithy:11]: I noticed that Message and MessageStatus are used in different security structures for output.So I decided to created this file to import this strings.

**Review by nhtruong — COMMENTED** [review]
🚀

**Review by nhtruong — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]
LGTM!

**Pull request merged**